app/javascript/gobierto_dashboards/webapp/visualizations/treeMap.vue

```diff
@@ -0,0 +1,368 @@
+      labelContractTotal: I18n.t('gobierto_dashboards.dashboards.visualizations.tooltip_treemap')
+    }
+  },
+  watch: {
+    data(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        this.dataNewValues = newValue
+        this.deepCloneData(newValue)
+        this.updateData = true
+      }
+    }
+  },
+  mounted() {
+    const containerChart = document.querySelector('.tree-map-container');
+    this.svgWidth = containerChart.offsetWidth;
+    this.dataTreeMapWithoutCoordinates = JSON.parse(JSON.stringify(this.data));
+    this.dataTooltips = JSON.parse(JSON.stringify(this.data));
+
+    this.dataForTooltips(this.dataTooltips)
+    this.transformDataTreemap(this.data)
+    this.resizeListener()
+  },
+  methods: {
+    handleTreeMapValue(value) {
+      if (this.selected_size === value) return;
+      this.selected_size = value
+      this.sizeForTreemap = value
+      if (this.updateData) {
+        this.deepCloneData(this.dataNewValues)
+      } else {
+        this.transformDataTreemap(this.data)
+      }
+    },
+    deepCloneData(data) {
+      const dataTreeMap = JSON.parse(JSON.stringify(data));
+      this.dataForTooltips(dataTreeMap)
+      this.transformDataTreemap(dataTreeMap)
+    },
+    transformDataTreemap(data) {
+      data.forEach(d => {
+        d.final_amount_no_taxes = +d.final_amount_no_taxes
+        d.value = 1
+      })
+      const finalAmountTotal = sumDataByGroupKey(data, 'contract_type', 'final_amount_no_taxes')
+      const countTotalContractsByType = sumDataByGroupKey(data, 'contract_type', 'value')
+
+      let dataContractsTreeMap = finalAmountTotal.map((item, i) => Object.assign({}, item, countTotalContractsByType[i]));
+
+      this.buildTreeMap(dataContractsTreeMap)
+
+    },
+    dataForTooltips(data) {
+      /* To avoid mixin different values which could be interferent in the treeMap,
+      it's better transform the data for tooltips in a different dataset */
+      const arrayValues = Array.from(new Set(data.map((d) => d.contract_type)))
+
+      this.dataForTableTooltip = this.operationsForTooltips(data, arrayValues)
+    },
+    operationsForTooltips(data, values) {
+      /* Create an array of objects, create an object for every type of contract
+      Inside, one key-value for mean, median, and the top five contracts by the final amount */
+      let subDataTooltip = []
+      for (let index = 0; index < values.length; index++) {
+        let element = values[index]
+        let tooltipData = {}
+        let dataFilter = data.filter(contract => contract.contract_type === element)
+
+        const amountsContractsArray = dataFilter.map(({ final_amount_no_taxes = 0 }) =>
+          parseFloat(final_amount_no_taxes)
+        );
+        const meanContracts = d3.mean(amountsContractsArray);
+        const medianContracts = d3.median(amountsContractsArray);
+
+        let sortDataContracts = dataFilter.sort((a, b) => a.final_amount_no_taxes < b.final_amount_no_taxes);
```

Comment:
Is this clause working in chrome? Sometimes, I had to set explicit `1` / `-1` as returning values